base/nvti.c

```diff
@@ -1057,6 +1057,46 @@ nvti_set_solution_type (nvti_t *n, const gchar *solution_type)
   return (0);
 }
 
+/**
+ * @brief Add a tag to the NVT tags.
+ *
+ * @param n The NVT Info structure.
+ *
+ * @param name  The tag name. A copy will be created from this.
+ * @param value The tag value. A copy will be created from this.
+ *
+ * @return 0 for success. Anything else indicates an error.
+ */
+int
+nvti_add_tag (nvti_t *n, const gchar *name, const gchar *value)
+{
+  char *new;
+
+  if (!n)
+    return (-1);
+
+  if (!name || !name[0])
+    return (-2);
+
+  if (!value || !value[0])
+    return (-3);
+
+  if (n->tag)
+    {
+      new = g_strconcat (n->tag, "|", name, "=", value, NULL);
+      g_free (n->tag);
+    }
+  else
+    new = g_strconcat (name, "=", value, NULL);
+
+  if (new && new[0])
```

Comment:
This seems wrong, if you don't have other changes in mind.

new will always have an actual value due to the checks and the two g_strconcat calls above.

I think new should be declared in the if branch and the else should just assign to n->tag.

Reply:
I condensed this now.

Reply:
Better, thanks.  newtag is only used in the 'if' branch, so you could
declare it in the 'if' branch, to make that clear.

Reply:
Done.